cmd/command.go

```diff
@@ -311,18 +404,110 @@ func hset(cli *Client, args [][]byte) (interface{}, error) {
 	return redcon.SimpleInt(count), nil
 }
 
-func hget(cli *Client, args [][]byte) (interface{}, error) {
+func hSetNX(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 3 {
+		return nil, newWrongNumOfArgsError("hsetnx")
+	}
+
+	key, field, value := args[0], args[1], args[2]
+	ok, err := cli.db.HSetNX(key, field, value)
+	if err != nil {
+		return nil, err
+	}
+	if ok {
+		return redcon.SimpleInt(1), nil
+	}
+	return redcon.SimpleInt(0), nil
+}
+
+func hGet(cli *Client, args [][]byte) (interface{}, error) {
 	if len(args) != 2 {
 		return nil, newWrongNumOfArgsError("hget")
 	}
 	val, err := cli.db.HGet(args[0], args[1])
 	return val, err
 }
 
+func hmGet(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) < 2 {
+		return nil, newWrongNumOfArgsError("hmget")
+	}
+	return cli.db.HMGet(args[0], args[1:]...)
+}
+
+func hDel(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) < 2 {
+		return nil, newWrongNumOfArgsError("hdel")
+	}
+	count, err := cli.db.HDel(args[0], args[1:]...)
+	return redcon.SimpleInt(count), err
+}
+
+func hExists(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 2 {
+		return nil, newWrongNumOfArgsError("hexists")
+	}
+	ok, err := cli.db.HExists(args[0], args[1])
+	if err != nil {
+		return nil, err
+	}
+	if ok {
+		return redcon.SimpleInt(1), nil
+	}
+	return redcon.SimpleInt(0), nil
+}
+
+func hLen(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 1 {
+		return nil, newWrongNumOfArgsError("hlen")
+	}
+	return redcon.SimpleInt(cli.db.HLen(args[0])), nil
+}
+
+func hKeys(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 1 {
+		return nil, newWrongNumOfArgsError("hkeys")
+	}
+	return cli.db.HKeys(args[0])
+}
+
+func hVals(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 1 {
+		return nil, newWrongNumOfArgsError("hvals")
+	}
+	return cli.db.HVals(args[0])
+}
+
+func hGetAll(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 1 {
+		return nil, newWrongNumOfArgsError("hgetall")
+	}
+	return cli.db.HGetAll(args[0])
+}
+
+func hStrLen(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 2 {
+		return nil, newWrongNumOfArgsError("hstrlen")
+	}
+	return redcon.SimpleInt(cli.db.HStrLen(args[0], args[1])), nil
+}
+
+func hScan(cli *Client, args [][]byte) (interface{}, error) {
```

Comment:
`scan` is a little more complicated.
In Redis, it contains a param `cursor` to scan, but our method doesn`t have it.

https://redis.io/commands/scan/

Reply:
OK, but you didn't add `cursor` in the `HScan` method. I just called the `HScan` method in this function.